sonic_platform_base/module_base.py

```diff
@@ -168,10 +170,12 @@ def reboot(self, reboot_type):
         Args:
             reboot_type: A string, the type of reboot requested from one of the
             predefined reboot types: MODULE_REBOOT_DEFAULT, MODULE_REBOOT_CPU_COMPLEX,
-            MODULE_REBOOT_FPGA_COMPLEX or MODULE_REBOOT_DPU
+            MODULE_REBOOT_FPGA_COMPLEX, MODULE_REBOOT_DPU or MODULE_REBOOT_SMARTSWITCH
```

Comment:
Why does the module need to know if other modules are rebooting as well? Do you expect anything different that needs to be done in the case of a chassis reboot?

Reply:
According to our last discussion, in the context of the DPU, it needs to determine whether it is undergoing an individual reboot or if the entire switch is rebooting. MODULE_REBOOT_DPU is the reboot type when individual DPU is undergoing reboot and MODULE_REBOOT_SMARTSWITCH is when DPU undergoing reboot as part of entire switch reboot.